library/std/src/sys/unix/thread.rs

```diff
@@ -13,6 +13,23 @@ pub const DEFAULT_MIN_STACK_SIZE: usize = 1024 * 1024;
 #[cfg(target_os = "vxworks")]
 pub const DEFAULT_MIN_STACK_SIZE: usize = 256 * 1024;
 
+#[cfg(target_os = "fuchsia")]
+mod zircon {
+    type zx_handle_t = u32;
+    type zx_status_t = i32;
+    pub const ZX_PROP_NAME: u32 = 3;
+
+    extern "C" {
+        pub fn zx_object_set_property(
+            handle: zx_handle_t,
+            property: u32,
+            value: *const libc::c_void,
+            value_size: libc::size_t,
+        ) -> zx_status_t;
+        pub fn zx_thread_self() -> zx_handle_t;
+    }
+}
```

Comment:
I think we should declare them on the libc crate so that we can keep consistency with other environments and it's also beneficial for the libc users.

Reply:
@JohnTitor I think it's out of scope for `libc` crate, because it's wrapper around `libc`, not the Zircon API.

We have already tiny declarations for Zircon API in [standard library](https://github.com/rust-lang/rust/blob/2ac0b3ed54b3ddfb2f6abfd7f0c1cbd6a924fc1c/library/std/src/sys/unix/process/zircon.rs)

Reply:
Ah, I see! Thanks for clarifying and the reference.